integration_tests/framework/common.go

```diff
@@ -0,0 +1,266 @@
+package integration_tests_framework
+
+import (
+	"crypto/rand"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	cliWrappers "github.com/konflux-ci/konflux-build-cli/pkg/cliwrappers"
+	l "github.com/konflux-ci/konflux-build-cli/pkg/logger"
+)
+
+const (
+	// Name of the CLI binary
+	KonfluxBuildCli = "konflux-build-cli"
+	// Directory where to put / expect the CLI binary to test
+	KonfluxBuildCliCompileDir = "/tmp"
+)
+
+var (
+	containerTool string
+	cliBinPath    string
+)
+
+func init() {
+	cliBinPath = path.Join(KonfluxBuildCliCompileDir, KonfluxBuildCli)
+
+	// Init logger
+	logLevel := "info"
+	logLevelEnv := os.Getenv("KBC_LOG_LEVEL")
+	if logLevelEnv != "" {
+		logLevel = logLevelEnv
+	}
+	if err := l.InitLogger(logLevel); err != nil {
+		fmt.Printf("failed to init logger: %s", err.Error())
+		os.Exit(2)
+	}
+
+	// Detect container tool to use
+	if ct := os.Getenv("KBC_TEST_CONTAINER_TOOL"); ct != "" {
+		containerTool = ct
+	} else if dockerInstalled, _ := cliWrappers.CheckCliToolAvailable("docker"); dockerInstalled {
+		containerTool = "docker"
+	} else if podmanInstalled, _ := cliWrappers.CheckCliToolAvailable("podman"); podmanInstalled {
+		containerTool = "podman"
```

Comment:
I figured out the cause of the problems. I have the `podman-docker` package installed, so my system does have the `docker` command, but it's an alias to podman. So the tests were doing the docker-specific workflow, but in my case it was actually podman 😄 

Prefering podman over docker should be a reasonable solution I think

```suggestion
	} else if podmanInstalled, _ := cliWrappers.CheckCliToolAvailable("podman"); podmanInstalled {
		containerTool = "podman"
	} else if dockerInstalled, _ := cliWrappers.CheckCliToolAvailable("docker"); dockerInstalled {
		containerTool = "docker"
```